perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -54,6 +80,107 @@
                   'num_machines': 1}
 
 
+JOB_FILE_TEMPLATE = """
```

Comment:
I see. Thanks yanting!

Noah, is it possible to not change the default behavior with your new code?
That is, by default, still use the old job files. And opt-in to use the new
template-based job files generation logic (doing random rw with varying
queue depths).



On Aug 20, 2015, at 10:03 PM, yuyanting <notifications@github.com> wrote:

In perfkitbenchmarker/benchmarks/fio_benchmark.py
<https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/421#discussion_r37606815>
:

> @@ -54,6 +80,107 @@
>                    'num_machines': 1}
>
>
> +JOB_FILE_TEMPLATE = """

@stfeng <https://github.com/stfeng> @noahl <https://github.com/noahl> The
default behavior of the new code is doing random RW with varying queue
depth, while the default behavior of the old code is doing seq RW, ran RW,
which is more like a overview of the disk performance.
If user is using their own fio.job, this PR won't make any difference.
However, if someone write some automation using the default old fio
benchmark, the published metric will definitely be different and we might
break their automation.
I suggest, we should at least mention in CHANGES.md that this is a breaking
change that changes the default behavior of the fio benchmark, or go
through the community, which is far more complex, since this benchmark is
in standard_set.

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/421/files#r37606815>
.